bundles/CoreBundle/Resources/migrations/Version20190403120728.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Pimcore\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Pimcore\Db;
+use Pimcore\Migrations\Migration\AbstractPimcoreMigration;
+
+class Version20190403120728 extends AbstractPimcoreMigration
+{
+    public function doesSqlMigrations(): bool
+    {
+        return true;
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        try {
+            $db = Db::get();
+
+            // add index to primary key in relations tables
+            $relationTables = $db->fetchAll("SHOW TABLES LIKE 'object\_relations\_%'");
+            foreach ($relationTables as $table) {
+                $relationTable = current($table);
+                $this->addSql('ALTER TABLE `' . $relationTable . '` ADD PRIMARY KEY `PRIMARYKEY` (`src_id`, `dest_id`, `ownertype`, `ownername`, `fieldname`, `type`, `position`, `index`), DROP INDEX `PRIMARY`');

```

Comment:
why not drop primary key first and then create new primary key with same name? something like that: 

``` 
alter table object_relations_33 drop primary key;

alter table object_relations_33
	add primary key (src_id, dest_id, ownertype, ownername, fieldname, type, position, `index`);
```